include/ipc/ipc_ringbuf.h

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_IPC_SERVICE_IPC_RINGBUF_H_
+#define ZEPHYR_INCLUDE_IPC_SERVICE_IPC_RINGBUF_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief IPC Service backend
+ * @ingroup ipc_service_api
+ * @{
+ */
+
+/**
+ * @brief Shared messaging ring buffer
+ *
+ * The ringbuf implements lightweight unidirectional buffer with read/write
+ * semantics on top of a memory region shared by the reader and writer. It
+ * embeds cache and memory barier management to ensure correct data access.
+ *
+ * This structure supports single writter and reader. Data stored in the buffer
+ * is encapsulated to a message with first byte reserved to contain message
+ * length. Single read allows to get the data written by the single write.
+ */
+struct ringbuf {
```

Comment:
Why? we have a ring buffer implementation already?

Reply:
@nordic-krch FYI

Reply:
> Why? we have a ring buffer implementation already?

Hi @carlescufi , there are changes specific to this implementation. If you think the name is confusing with current code we can change it. The `mring` maybe (from messages ring)?

Reply:
I just updated it with new convention. The memory access abstraction is renamed to: `icmsg_buf` (inter core messaging buffer). The backend that relies on this is renamed to `ipc_icmsg`.